workflow-pacta-cop.Rmd

```diff
@@ -0,0 +1,516 @@
+
+- PACTA_Analysis
+
+- create_interactive_report
+
+- pacta-data
+
+- StressTestingModelDev
+
+
+```{r}
+
+library(fs)
+library(readr)
+library(yaml)
+library(dplyr)
+
+start_dir <- getwd()
+
+local_git_dir <- "C:/Users/clare/Desktop/PACTA"
+
+test_investor <- unique(portfolio_file$Investor.Name)[1]
+test_portfolio <- portfolio_file %>% 
+  filter(Investor.Name == test_investor) %>% 
+  select(Portfolio.Name) %>%
+  distinct() %>% pull()
+test_portfolio <- test_portfolio[1]
+test_peergroup <-unique(portfolio_file$Peer.Group)[1]
+
+portfolio_data <- portfolio_file %>% 
+  filter(Investor.Name == test_investor,
+         Portfolio.Name == test_portfolio) %>% 
+  select(-Peer.Group)
+
+
+portfolio_name <- test_portfolio
+investor_name <- test_investor
+peer_group_type <- test_peergroup
+language_to_print <- languages[1]
+
+yaml_data <-
+  list(default = list(
+    parameters =
+      list(
+        portfolio_name_in = portfolio_name,
+        investor_name_in = investor_name,
+        peer_group = peer_group_type,
+        language = language_to_print,
+        project_code = project_code
+      )
+  ))
+
+out_dir <- path(local_git_dir, "PACTA_analysis", "working_dir")
+
+unlink(out_dir, recursive = T, force = T)
+dir.create(path(out_dir, "00_Log_Files"), recursive = T)
+dir.create(path(out_dir, "10_Parameter_File"), recursive = T)
+dir.create(path(out_dir, "20_Raw_Inputs"), recursive = T)
+dir.create(path(out_dir, "30_Processed_Inputs"), recursive = T)
+dir.create(path(out_dir, "40_Results"), recursive = T)
+dir.create(path(out_dir, "50_Outputs"), recursive = T)
+
+write_yaml(yaml_data, fs::path(out_dir, "10_Parameter_File", paste0(test_investor, "_PortfolioParameters.yml")), indent = 4)
+
+portfolio_name_ref_all <- test_investor
+portfolio_root_dir <- out_dir
+
+cli::cli_h1(paste0("running for: ", portfolio_name_ref_all, " (test #", test_investor, ")"))
+
+portfolio_root_dir <- "working_dir"
+
+if (nrow(portfolio_raw_data) > 0) {
+  print("port rows > 0")
+  
+  write_csv(data.frame(portfolio_data), path(out_dir, "20_Raw_Inputs", paste0(test_investor, ".csv")))
+  
+  try(source("web_tool_script_1.R"))
+  
+  try(source("web_tool_script_2.R"))
+}
+
+try(source("web_tool_script_3.R"))
+
+```
+
+The output report can be found in `{r} path(out_dir, "50_Outputs")`
+
+Feel free to repeast this process, changing the test_investor or test_portfolio. 
+
+## Docker Image
+
+Once this process has succeeded and you are happy with the contents of the report and the peers results look realistic. It is time to generate the docker image. 
+
+First, push all outstanding commits, and ensure that you have created appropriate PRs in all repos. 
+
+When done, alert the tech team, and flag it to the pacta-cop Slack channel to understand any other changes that could be occurring at the same time. 
+
+This project should be added to the docker image to ensure that it is tested for critical failure with every new release of the online tool.  
```

Comment:
not sure I understand what this sentence means

Reply:
I hope I've cleared that up. But basically we should be testing all active projects with each docker image release.